woodwork/statistics_utils.py

```diff
@@ -343,6 +343,104 @@ def _get_value_counts(dataframe, ascending=False, top_n=10, dropna=False):
     return val_counts
 
 
+def _get_box_plot_info_for_column(series, quantiles=None):
+    """Gets the information necessary to create a box and whisker plot with outliers for a numeric column
+        using the IQR method.
+
+    Args:
+        series (Series): Data for which the box plot and outlier information will be gathered.
+            Will be used to calculate quantiles if none are provided.
+        quantiles (dict[float -> float], optional): A dictionary containing the quantiles for the data
+            where the key indicates the quantile, and the value is the quantile's value for the data.
+
+    Note:
+        The minimum quantiles necessary for outlier detection using the IQR method are the
+        first quartile (0.25) and third quartile (0.75). If these keys are missing from the quantiles dictionary,
+        the following quantiles will be calculated: {0.0, 0.25, 0.5, 0.75, 1.0}, which correspond to
+        {min, first quantile, median, third quantile, max}.
+
+    Returns:
+        (dict[str -> float,list[number]]): Returns a dictionary containing box plot information for the Series.
+            The following elements will be found in the dictionary:
+
+            - low_bound (float): the lower bound below which outliers lay - to be used as a whisker
+            - high_bound (float): the high bound above which outliers lay - to be used as a whisker
+            - quantiles (list[float]): the quantiles used to determine the bounds.
+                If quantiles were passed in, will contain all quantiles passed in. Otherwise, contains the five
+                quantiles {0.0, 0.25, 0.5, 0.75, 1.0}.
+            - low_values (list[float, int]): the values of the lower outliers
+            - high_values (list[float, int]): the values of the upper outliers
+            - low_indices (list[int]): the corresponding index values for each of the lower outliers
+            - high_indices (list[int]): the corresponding index values for each of the upper outliers
+    """
+    if not series.ww._schema.is_numeric:
+        raise TypeError('Cannot calculate box plot statistics for non-numeric column')
+
+    if quantiles and not isinstance(quantiles, dict):
+        raise TypeError('quantiles must be a dictionary.')
+
+    if dd and isinstance(series, dd.Series):
+        series = series.compute()
+    if ks and isinstance(series, ks.Series):
+        series = series.to_pandas()
```

Comment:
I'm again thinking about my previous company where we were always trying to be extremely mindful of what sorts of algorithms and techniques were viable in a distributed computing (big data) context.  I've been told we're still a ways off from really targeting that scale.  So I don't think it's worth holding up this PR.  However, I thought I'd offer some more thoughts on it here to hopefully get us thinking about these questions.

Both Dask and Koalas offer `quantile` methods that use efficient approximations to obtain quantile figures in-cluster.  If `series` were a Dask or Koalas object, we could call out to the respective method.  It might also be required to do something like `.compute()` or `.to_pandas()` to force the computation to execute inside this method in order to get quantile figures that can be used for filtering later in the method<sup>[1](#f1)</sup>.  But then the filtering itself could also be done through the Dask or Koalas API and we could eventually just pass back the resulting Dask or Koalas object.  Then, the caller could decide if it wants to do the computation right then and there to get the concrete outliers or wait until a later time.

In a big data context, it would really be much more ideal if this method just returned the filtered series object (which would be either a Dask or Koalas object) instead of forcing the values into a python list.  But I guess that includes some details about serialization required to move things between services.  I'm less familiar with that part of things right now so I don't have much to say about it.  In general, I think it's just better if methods are about one thing and don't include implicit details about the context from which they're being called (like requirements of eventual serialization of the result).

<b id="f1">1</b>: As a side note, if we do make a synchronous cluster call mid-function, that's not that great.  It's always better if you can defer the distributed computation until some time when it's expected that you might call out to a cluster and wait around a long time for a result to come back.  If you're mixing synchronous cluster calls into a utility function like this, that will make the function a lot less useful.

Reply:
would it be worth adding some of these concerns to #1045? We can start compiling some of the areas where we're maybe not the happiest with how we've handled dask and koalas there.

Reply:
Yeah, I suppose so.  I'll add a comment there referring to this one.